File: Csharp-MsDependencyInjection/LcdNumbers/LcdDisplay.cs
using System.Collections.Generic;

namespace Org.Codecop.Lcdnumbers
{
    /// <summary>Combines Converter, Scaler and Printer.</summary>
    public class LcdDisplay : ILcdDisplay
    {
        private readonly IDigitsSplitter digitsSplitter;
        private readonly IDigitScaler digitScaler;
        private readonly IDigitPrinter digitPrinter;

        public LcdDisplay(IDigitsSplitter digitsSplitter, IDigitScaler digitScaler, IDigitPrinter digitPrinter)
        {
          this.digitsSplitter = digitsSplitter;
          this.digitScaler = digitScaler;
          this.digitPrinter = digitPrinter;
        }

        public virtual string ToLcd(int number, Scaling scaling)
        {
            IList<Digit> digits = digitsSplitter.Convert(number);
            IList<Digit> scaled = digitScaler.Scale(digits, scaling);
            return digitPrinter.Render(scaled);
        }
    }
}

File: Java-Spring/src/main/java/org/codecop/lcdnumbers/LcdDisplay.java
package org.codecop.lcdnumbers;

import java.util.List;

/**
 * Combines Converter, Scaler and Printer.
 */
public class LcdDisplay {

    private final DigitsSplitter digitsSplitter;
    private final DigitScaler digitScaler;
    private final DigitPrinter digitPrinter;

    public LcdDisplay(DigitsSplitter digitsSplitter, DigitScaler digitScaler, DigitPrinter digitPrinter) {
        this.digitsSplitter = digitsSplitter;
        this.digitScaler = digitScaler;
        this.digitPrinter = digitPrinter;
    }

    public String toLcd(int number, Scaling scaling) {
        List<Digit> digits = digitsSplitter.convert(number);
        List<Digit> scaled = digitScaler.scale(digits, scaling);
        return digitPrinter.render(scaled);
    }
}
